src/main/java/cuchaz/enigma/Enigma.java

```diff
@@ -56,7 +55,18 @@ public EnigmaProject openJar(Path path, ProgressListener progress) throws IOExce
 
 		services.get(JarIndexerService.TYPE).forEach(indexer -> indexer.acceptJar(classCache, jarIndex));
 
-		return new EnigmaProject(this, classCache, jarIndex);
+		MessageDigest digest;
+		try {
+			digest = MessageDigest.getInstance("MD5");
```

Comment:
SHA-256 is unnecessary for a file checksum that you could easily fake if you for some reason wanted to. The jar checksum is only checked to prevent people from accidentally logging in with the wrong jar.

Reply:
@Earthcomputer I think you misunderstood coderbot's explanation. But @coderbot16 so is MD5.

Reply:
I think what he was trying to say is that we should use the strongest algorithm that's guaranteed to be supported. But I disagree because the strength is not what's important here, it's preventing accidental logins with the wrong Jar.
Just because it's tractable to generate files with the same MD5, doesn't mean it's likely at all to happen accidentally, which is really what we care about here. There's no reason to use a hash that is many times slower to compute.

Reply:
In that case here you go: SHA-1 is faster: https://automationrhapsody.com/md5-sha-1-sha-256-sha-512-speed-performance/

Reply:
Fair enough, I can switch it to SHA-1